### PR TITLE
br: cli refactor backup error handling logic

### LINKED ISSUE #54696: br: refactor br cli backup error handling logic
## Enhancement
The current backup error handling logic is located in the retry file, better to put it into a separate file for readability since the retry file is shared by a few other use cases as well.

### PR DESCRIPTION
**Pull request by Tristan1900**
### What problem does this PR solve?
Issue Number: close #54696

Problem Summary:

### What changed and how does it work?
1. Moved backup specific error handling logic out of the retry file, since the retry is shared by other usages as well.
2. Renamed the method and a few other variables. 
3. Move as much as possible of hardcoded strings to be const so unit test can use
4. Delete unused methods, `handleErrorPb` and `HandleError `
5. Delete and move the permission file, as in the file only one method is related to permission.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note
```release-note
None
```

### CODE REVIEW

File: br/pkg/utils/error_handling.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	defer ec.mu.Unlock()
+	ec.encounterTimes[uuid]++
+	if ec.encounterTimes[uuid] <= ec.encounterTimesLimitation {
+		return ErrorHandlingResult{Retry, retryOnUnknownErrorMsg}
+	}
+	return ErrorHandlingResult{GiveUp, noRetryOnUnknownErrorMsg}
+}
+
+// messageIsNotFoundStorageError checks whether the message returning from TiKV is "NotFound" storage I/O error
+func messageIsNotFoundStorageError(msg string) bool {
```
Comment (Tristan1900) [inline br/pkg/utils/error_handling.go:160]: @3pointer does the storage here mean S3 or any backup destination? At first I thought it's TiKV but there is a proto type for it already so got a bit confused. Thanks

File: br/pkg/utils/error_handling.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
+package utils
```
Comment (3pointer) [inline br/pkg/utils/error_handling.go:1]: new file should add license first, you can copy from other files.

File: br/pkg/utils/error_handling.go (hunk 1 above)
Comment (3pointer) [reply br/pkg/utils/error_handling.go:160]: at this context storage means TiKV.

File: br/pkg/utils/error_handling.go (hunk 2 above)
Comment (Tristan1900) [reply br/pkg/utils/error_handling.go:1]: ah! good catch

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

File: br/pkg/utils/error_handling.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+type ErrorHandlingResult struct {
+	Strategy ErrorHandlingStrategy
+	Reason   string
+}
+
+type ErrorHandlingStrategy int
+
+const (
+	// Retry error can be retried but will consume the backoff attempt quota.
+	Retry ErrorHandlingStrategy = iota
```
[the remark this answers is not in the document]
Comment (Tristan1900) [reply br/pkg/utils/error_handling.go:76]: good point on the namespace, let me rename it to something clearer

**Review by YuJuncen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**